source/tutorials/packaging-existing-software.md

```diff
@@ -0,0 +1,870 @@
+
+To enter a useful development shell, we'll pass the dependencies from `nativeBuildInputs` and `buildInputs` to `nix-shell -p`. We'll also make sure to include `git`, so we can clone the `icat` GitHub repository:
+
+```console
+$ nix-shell -p pkg-config imlib2 xorg.libX11.dev git
+```
+
+After many lines of output about Nix copying dependencies, we can use the following commands to retrieve and build the `icat` source code:
+
+```console
+$ git clone https://github.com/atextor/icat
+$ cd icat
+$ make
+```
+
+In the current `master` branch of `icat`, a warning is thrown when building:
+
+```console
+[nix-shell:~/icat]$ make
+gcc -c -Wall -pedantic -std=c99 -D_DEFAULT_SOURCE `pkg-config --cflags imlib2` -o icat.o icat.c
+icat.c: In function ‘main’:
+icat.c:319:33: warning: ignoring return value of ‘write’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
+  319 |                                 write(tempfile, &buf, 1);
+      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
+gcc -o icat icat.o  `pkg-config --libs imlib2`
+```
+
+However, this does not prevent the binary from being produced; an `icat` executable is now present in the local directory, and it's up to us to decide what to do with it.
+
+### installPhase
+In order to make packages available for other packages to depend on, Nix copies everything to the Nix store (at `/nix/store`), and symlinks them from there into build contexts and development environments.
+
+The `Makefile` doesn't provide an installation step, so we must produce one for our derivation, using the [`installPhase` attribute](https://nixos.org/manual/nixpkgs/stable/#ssec-install-phase), which contains a list of command strings to execute to accomplish the installation.
+
+The `icat` executable is only used at runtime, and isn't a compile-time input for anything else at this point, so we only need to concern ourselves with the `bin` output. In Nix, the result of a build is copied to a location stored in the `$out` variable accessible in the derivation's component scripts; we'll create a `bin` directory within that, and then copy our `icat` executable there:
+
+```nix
+# icat.nix
+{ pkgs
+, lib
+, stdenv
+}:
+
+stdenv.mkDerivation {
+  name = "icat";
+  src = pkgs.fetchFromGitHub {
+    owner = "atextor";
+    repo = "icat";
+    rev = "master";
+    sha256 = "sha256-b/2mRzCTyGkz2I1U+leUhspvW77VcHN7Awp+BVdVNRM=";
+  };
+
+  nativeBuildInputs = with pkgs; [ pkg-config ];
+  buildInputs = with pkgs; [ imlib2 xorg.libX11.dev ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp icat $out/bin
+  '';
+}
+```
+
+After running our `nix-build` command one last time, we can `ls` in the local directory to find a `result` symlink to the Nix store, with `result/bin/icat` the executable we built. Success!
+
+## Contributing our Work
```

Comment:
A tutorial on making good contributions to Nixpkgs is a high value target, but it would open a real can of worms here. Especially we'd have to clarify what should go into the manual (and how to structure it there) and what should be part of a step-by-step tutorial. The other thing are the million little details that we'd have to get right to make sure this procedure will in fact lead to success. That should absolutely involve Nixpkgs maintainers. I already see a handful of things that won't pass review, but let's not get into that and instead split that part out for now and focus on polishing the tutorial to get as close to what's needed as possible, avoiding additional steps later (e.g. not grab the HEAD of master by default).